app/models/ability.rb

```diff
@@ -0,0 +1,17 @@
+class Ability
+  include CanCan::Ability
+  def initialize(user)
+    user ||= User.new
+    can :read, :all
+    can :create, :user
+
+    can :manage, :all if user.role == 'admin'
+
+    return unless user.role == 'user'
+
+    can :create, Post
+    can :create, Comment
+    can :destroy, Comment
+    can :destroy, Post
```

Comment:
- [ ] Please ensure that only authors of Posts and Comments can destroy them. At the moment, you're only hiding the UI buttons for these actions. I suggest something like this: 🤓 🔧 

```rb
# owner of post can perform all view, update or destroy own post if Post.author == user
         can :manage, Post, author: user 
# owner of comment can perform all view, update or destroy own comment if Comment.author == user
         can :manage, Comment, author: user
```

Reply:
@anagudelogu Got it! thank you for that advise, I've already made it